### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by cjweir**
A first pass at refactoring the code and replacing the TCV with the EM one.

### CODE REVIEW

**Review by jc-healy — APPROVED** [review]

File: vectorizers/token_cooccurrence_vectorizer.py [hunk 1]
```diff
@@ -1441,7 +1441,14 @@ def reduce_dimension(self, dimension=150, algorithm="arpack", n_iter=10):
...
 
-        u, s, v = svds(self.reduced_matrix_, k=dimension)
+        if algorithm == "arpack":
+            u, s, v = svds(self.reduced_matrix_, k=dimension)
+        elif algorithm == "randomized":
+            u, s, v = randomized_svd(self.reduced_matrix_, n_components=dimension, n_iter=n_iter)
+        else:
+            raise ValueError("algorithm should be one of 'arpack' or 'randomized'")
+
+        u, v = svd_flip(u, v)
```
[the remark this answers is not in the document]
Comment (cjweir) [reply vectorizers/token_cooccurrence_vectorizer.py]: It was Leland's piece of code... I just merged it in

File: vectorizers/ngram_vectorizer.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
+"""
+This is a module to be used as a reference for building other modules
```
Comment (lmcinnes) [inline vectorizers/ngram_vectorizer.py:2]: Module docstring can be updated

File: vectorizers/preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,646 @@
+"""
+This is a module to be used as a reference for building other modules
```
Comment (lmcinnes) [inline vectorizers/preprocessing.py:2]: Module docstring can be updated

**Comment by lmcinnes** [comment]
Just some minor auto-generated module docstrings (which may have been there already). It's not critical, but we may as well fix them while we can.

**Pull request merged**